pkg/commands/run.go

```diff
@@ -234,6 +234,7 @@ func (e *Executor) getConfigForCommandLine() (*config.Config, error) {
 	// Use another config variable here, not e.cfg, to not
 	// affect main parsing by this parsing of only config option.
 	initFlagSet(fs, &cfg, e.DBManager, false)
+	initVersionFlagSet(fs, &cfg)
```

Comment:
can you explain rationale behind this line ?

Reply:
Sure. We need to refactor this code. Now the executer uses same set of the flags for all commands, and when I defined the flag for version command only it's failed with an error with unknown `--format` arg, here: https://github.com/golangci/golangci-lint/blob/f7417e6ff048c50b89364a074881a0dddec5a5ac/pkg/commands/run.go#L244

This is kind of workaround, I think we need to have a hard refactoring of all this code and do not mix the flags for `run` command with other command, ideally every command must have independent sets of flags

Reply:
Oh I see :+1: